kairon/shared/data/processor.py

```
@@ -8957,6 +8960,118 @@
+        from ..channels.whatsapp.bsp.factory import BusinessServiceProviderFactory
+
+        media_id = None
+        try:
+            bsp_type= WhatsappBSPTypes.bsp_360dialog.value
+            media_id = await BusinessServiceProviderFactory.get_instance(bsp_type).upload_media_file(bot, bsp_type, user,
+                                                                file_info.filename, file_info.content_type,
+                                                                file_info.size)
+            logger.info(f"Media uploaded successfully: {media_id}")
+        except Exception as e:
+            logger.error(f"Error uploading file to BSP: {str(e)}")
+            raise AppException(f"Media upload failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)